### PR TITLE
libnbd: cve-2024-7383 (mariner 2.0)

### PR DESCRIPTION
**Pull request by bfjelds**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 
Create patches to fix libnbd CVE-2024-7383

###### Change Log  
- Create 3 patch files
- Update spec version

###### Does this affect the toolchain?  
**NO**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2024-7383

###### Test Methodology
- https://dev.azure.com/mariner-org/mariner/_build/results?buildId=624653&view=results
- with single combined patch: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=625051&view=results
- fasttrack/2.0: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=633963&view=results

### CODE REVIEW

File: SPECS/libnbd/CVE-2024-7383-1.patch [hunk 1]
```diff
@@ -0,0 +1,58 @@
+From c6cc19319f39c09c4ff74b47101eb217e75d3b43 Mon Sep 17 00:00:00 2001
```
Comment (mfrw) [inline SPECS/libnbd/CVE-2024-7383-1.patch:1]: How about we combine all the patch hunks in a single patch:
```bash
cat CVE-2024-7383-* > CVE-2024-7383.patch
```

Comment (bfjelds) [reply SPECS/libnbd/CVE-2024-7383-1.patch:1]: sure ... wasn't sure how to combine patches, but will try your suggestion.

Comment (bfjelds) [reply SPECS/libnbd/CVE-2024-7383-1.patch]: combined into one.

**Review by mfrw — APPROVED** [review]

**Review by jslobodzian — APPROVED** [review]

**Comment by PawelWMS** [comment]
Ignoring the Go test coverage test - it's failing due to GitHub's policy check, not the changes in the PR.

The single ptest failure comes from `nbdkit`, which is a known issue and not a regression.

**Pull request merged**